packages/session/internal/usecase/session.go

```diff
@@ -0,0 +1,93 @@
+package usecase
+
+import (
+	"catworks/luna/session/internal/config"
+	"catworks/luna/session/internal/domain"
+	"context"
+	"github.com/google/uuid"
+	"time"
+)
+
+type sessionUseCaseImpl struct {
+	cfg *config.Config
+
+	sessionStorage domain.SessionStorage
+	jwtService     domain.JWTService
+}
+
+func NewSessionUseCase(cfg *config.Config, sessionStorage domain.SessionStorage, jwtService domain.JWTService) domain.SessionUseCase {
```

Comment:
To avoid dragging around a long list of arguments, it makes sense to use [this](https://github.com/google/wire/blob/main/docs/best-practices.md#options-structs)

Reply:
Done